storm-core/src/jvm/org/apache/storm/daemon/supervisor/SupervisorData.java

```diff
@@ -0,0 +1,234 @@
+    private String supervisorId;
+    private String assignmentId;
+    private String hostName;
+    // used for reporting used ports when heartbeating
+    private AtomicReference<Map<Long, LocalAssignment>> currAssignment;
+    private StormTimer heartbeatTimer;
+    private StormTimer eventTimer;
+    private StormTimer blobUpdateTimer;
+    private Localizer localizer;
+    private AtomicReference<Map<String, Map<String, Object>>> assignmentVersions;
+    private AtomicInteger syncRetry;
+    private final Object downloadLock = new Object();
+    private AtomicReference<Map<String, List<ProfileRequest>>> stormIdToProfileActions;
+    private ConcurrentHashSet<String> deadWorkers;
+    private final IWorkerManager workerManager;
+
+    public SupervisorData(Map conf, IContext sharedContext, ISupervisor iSupervisor) {
+        this.conf = conf;
+        this.sharedContext = sharedContext;
+        this.iSupervisor = iSupervisor;
+        this.active = true;
+        this.upTime = Utils.makeUptimeComputer();
+        this.stormVersion = VersionInfo.getVersion();
+        this.workerThreadPids = new ConcurrentHashMap<String, String>();
+        this.deadWorkers = new ConcurrentHashSet();
+
+        List<ACL> acls = null;
+        if (Utils.isZkAuthenticationConfiguredStormServer(conf)) {
+            acls = SupervisorUtils.supervisorZkAcls();
+        }
+
+        try {
+            this.stormClusterState = ClusterUtils.mkStormClusterState(conf, acls, new ClusterStateContext(DaemonType.SUPERVISOR));
+        } catch (Exception e) {
+            LOG.error("supervisor can't create stormClusterState");
+            throw Utils.wrapInRuntime(e);
+        }
+
+        try {
+            this.localState = ConfigUtils.supervisorState(conf);
+            this.localizer = Utils.createLocalizer(conf, ConfigUtils.supervisorLocalDir(conf));
+        } catch (IOException e) {
+            throw Utils.wrapInRuntime(e);
+        }
+        this.supervisorId = iSupervisor.getSupervisorId();
+        this.assignmentId = iSupervisor.getAssignmentId();
+
+        try {
+            this.hostName = Utils.hostname(conf);
+        } catch (UnknownHostException e) {
+            throw Utils.wrapInRuntime(e);
+        }
+
+        this.currAssignment = new AtomicReference<Map<Long, LocalAssignment>>(new HashMap<Long,LocalAssignment>());
+
+        this.heartbeatTimer = new StormTimer(null, new DefaultUncaughtExceptionHandler());
+
+        this.eventTimer = new StormTimer(null, new DefaultUncaughtExceptionHandler());
+
+        this.blobUpdateTimer = new StormTimer("blob-update-timer", new DefaultUncaughtExceptionHandler());
+
+        this.assignmentVersions = new AtomicReference<Map<String, Map<String, Object>>>(new HashMap<String, Map<String, Object>>());
+        this.syncRetry = new AtomicInteger(0);
+        this.stormIdToProfileActions = new AtomicReference<Map<String, List<ProfileRequest>>>(new HashMap<String, List<ProfileRequest>>());
```

Comment:
Very minor, this should be `stormIdToProfilerActions`  There is an r at the end or Profile